src/pthread/thread.cpp

```diff
@@ -437,25 +437,23 @@ namespace boost
 
                 if (ts >  detail::timespec_duration::zero())
                 {
-    #   if defined(BOOST_HAS_PTHREAD_DELAY_NP) && !defined(BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC)
+                  // Use pthread_delay_np or nanosleep whenever possible here in the no_interruption_point
+                  // namespace because they do not provide an interruption point.
+    #   if defined(BOOST_HAS_PTHREAD_DELAY_NP)
     #     if defined(__IBMCPP__) ||  defined(_AIX)
                   BOOST_VERIFY(!pthread_delay_np(const_cast<timespec*>(&ts.get())));
     #     else
                   BOOST_VERIFY(!pthread_delay_np(&ts.get()));
     #     endif
-    #   elif defined(BOOST_HAS_NANOSLEEP) && !defined(BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC)
+    #   elif defined(BOOST_HAS_NANOSLEEP)
                   nanosleep(&ts.get(), 0);
-    #   elif defined(BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC)
+    #   else
+                  // Fall back to using a condition variable even though it does provide an interruption point.
```

Comment:
IIUC, this function is called only in the BOOST_THREAD_SLEEP_FOR_IS_STEADY case and so the #else never happens, isn't it?

Reply:
Yes, I believe that is correct. The #else should never get executed.